domenetjenester/beregning-ytelse/src/main/java/no/nav/ung/sak/ytelse/beregning/UngdomsytelseUtledTilkjentYtelse.java

```diff
@@ -0,0 +1,83 @@
+package no.nav.ung.sak.ytelse.beregning;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Period;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.inject.Inject;
+import no.nav.fpsak.tidsserie.LocalDateSegment;
+import no.nav.fpsak.tidsserie.LocalDateSegmentCombinator;
+import no.nav.fpsak.tidsserie.LocalDateTimeline;
+import no.nav.ung.kodeverk.ungdomsytelse.uttak.UngdomsytelseUttakAvslagsårsak;
+import no.nav.ung.sak.behandlingslager.ytelse.UngdomsytelseGrunnlag;
+import no.nav.ung.sak.behandlingslager.ytelse.UngdomsytelseGrunnlagRepository;
+import no.nav.ung.sak.behandlingslager.ytelse.sats.UngdomsytelseSatser;
+import no.nav.ung.sak.behandlingslager.ytelse.uttak.UngdomsytelseUttak;
+import no.nav.ung.sak.domene.typer.tid.DatoIntervallEntitet;
+import no.nav.ung.sak.ytelse.DagsatsOgUtbetalingsgrad;
+
+@Dependent
+public class UngdomsytelseUtledTilkjentYtelse implements UtledTilkjentYtelse {
+
+    private UngdomsytelseGrunnlagRepository ungdomsytelseGrunnlagRepository;
+
+    protected UngdomsytelseUtledTilkjentYtelse() {
+        // for proxy
+    }
+
+    @Inject
+    public UngdomsytelseUtledTilkjentYtelse(UngdomsytelseGrunnlagRepository ungdomsytelseGrunnlagRepository) {
+        this.ungdomsytelseGrunnlagRepository = ungdomsytelseGrunnlagRepository;
+    }
+
+    @Override
+    public Optional<List<TilkjentYtelsePeriode>> utledTilkjentYtelsePerioder(Long behandlingId) {
```

Comment:
Kan vi forenkle til List<TilkjentYtelsePeriode>, eller har vi behov for å skille tom liste fra tom optional? 

Kanskje enda bedre å kunne returnere en tidslinje (jeg er fan av det over liste av perioder, siden tidlinjer garanterer å ha håndtert overlapp), men det er kjedelig om vi må innføre nye klasser da

Reply:
Det er for å kunne skilje avslag fra ikkje fastsatt uttak, men eg veit ikkje om det er nødvendig